rig/machine_control/machine_controller.py

```diff
@@ -917,6 +935,30 @@ def send_signal(self, signal, app_id=Required):
         self._send_scp(0, 0, 0, SCPCommands.signal, arg1, arg2, arg3)
 
     @ContextMixin.use_contextual_arguments
+    def app_start(self, app_id=Required):
+        """Transmit a signal to start the application."""
+        self.send_signal(consts.AppSignal.start, app_id)
+
+    @ContextMixin.use_contextual_arguments
+    def app_stop(self, app_id=Required):
+        """Transmit a signal to stop the application."""
+        self.send_signal(consts.AppSignal.stop, app_id)
+
+    @ContextMixin.use_contextual_arguments
+    def sync0(self, app_id=Required):
```

Comment:
Something like that. Though a "wait for count" seems like a frequently useful thing, I can see it as being desirable to also check for failed processes etc. rather than waiting for a timeout or implement some other more customised timeout mechanism. It would also be useful for people using sync0 and 1 as general purpose application barriers.